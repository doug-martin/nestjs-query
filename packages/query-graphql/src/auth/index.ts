export * from './authorizer';
export * from './default-crud.authorizer';
export * from './tokens';
