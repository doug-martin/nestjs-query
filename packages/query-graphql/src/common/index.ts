export { DTONamesOpts, getDTONames, DTONames, getDTOIdTypeOrDefault } from './dto.utils';
export * from './external.utils';
export * from './resolver.utils';
export * from './object.utils';
