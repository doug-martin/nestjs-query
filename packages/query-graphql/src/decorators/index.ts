export {
  FilterableField,
  FilterableFieldOptions,
  FilterableFieldDescriptor,
  getFilterableFields,
} from './filterable-field.decorator';
export { ResolverMethodOpts } from './resolver-method.decorator';
export {
  CursorConnection,
  FilterableCursorConnection,
  OffsetConnection,
  FilterableOffsetConnection,
  UnPagedRelation,
  FilterableUnPagedRelation,
  Relation,
  FilterableRelation,
  RelationDecoratorOpts,
  RelationTypeFunc,
  getRelations,
} from './relation.decorator';
export * from './resolver-mutation.decorator';
export * from './resolver-query.decorator';
export * from './resolver-field.decorator';
export { Reference, ReferenceDecoratorOpts, ReferenceTypeFunc } from './reference.decorator';
export { ResolverSubscription, SubscriptionResolverMethodOpts } from './resolver-subscription.decorator';
export { InjectPubSub } from './inject-pub-sub.decorator';
export * from './skip-if.decorator';
export * from './aggregate-query-param.decorator';
export * from './hook.decorator';
export * from './decorator.utils';
export * from './hook-args.decorator';
export * from './authorizer.decorator';
export * from './inject-authorizer.decorator';
export * from './inject-custom-authorizer.decorator';
export * from './key-set.decorator';
export * from './authorize-filter.decorator';
export * from './query-options.decorator';
export * from './id-field.decorator';
