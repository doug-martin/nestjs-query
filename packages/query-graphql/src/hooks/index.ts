export * from './hooks';
export { createDefaultHook } from './default.hook';
export * from './tokens';
export * from './types';
