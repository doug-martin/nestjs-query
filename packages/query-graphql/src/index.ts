export * from './types';
export {
  FilterableField,
  FilterableFieldOptions,
  ResolverMethodOpts,
  Relation,
  FilterableRelation,
  CursorConnection,
  FilterableCursorConnection,
  OffsetConnection,
  FilterableOffsetConnection,
  UnPagedRelation,
  FilterableUnPagedRelation,
  RelationTypeFunc,
  RelationDecoratorOpts,
  Reference,
  ReferenceTypeFunc,
  ReferenceDecoratorOpts,
  InjectPubSub,
  BeforeCreateOne,
  BeforeCreateMany,
  BeforeUpdateOne,
  BeforeUpdateMany,
  BeforeDeleteOne,
  BeforeDeleteMany,
  BeforeQueryMany,
  BeforeFindOne,
  InjectAuthorizer,
  InjectCustomAuthorizer,
  Authorize,
  AuthorizerFilter,
  RelationAuthorizerFilter,
  HookArgs,
  MutationHookArgs,
  KeySet,
  QueryOptions,
  IDField,
  IDFieldOptions,
} from './decorators';
export * from './resolvers';
export * from './federation';
export { DTONamesOpts } from './common';
export { NestjsQueryGraphQLModule } from './module';
export { AutoResolverOpts } from './providers';
export { pubSubToken, GraphQLPubSub } from './subscription';
export { Authorizer, CustomAuthorizer, AuthorizerOptions, AuthorizationContext, OperationGroup } from './auth';
export {
  Hook,
  HookTypes,
  BeforeCreateOneHook,
  BeforeCreateManyHook,
  BeforeUpdateOneHook,
  BeforeUpdateManyHook,
  BeforeDeleteOneHook,
  BeforeDeleteManyHook,
  BeforeQueryManyHook,
  BeforeFindOneHook,
} from './hooks';
export { AuthorizerInterceptor, AuthorizerContext, HookInterceptor, HookContext } from './interceptors';
