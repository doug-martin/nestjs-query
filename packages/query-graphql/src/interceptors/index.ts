export * from './hook.interceptor';
export * from './authorizer.interceptor';
