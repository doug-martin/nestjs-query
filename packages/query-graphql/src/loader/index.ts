export * from './find-relations.loader';
export * from './query-relations.loader';
export * from './count-relations.loader';
export * from './aggregate-relations.loader';
export * from './dataloader.factory';
