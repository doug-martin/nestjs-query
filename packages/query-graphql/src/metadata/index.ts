import { GraphQLQueryMetadataStorage } from './metadata-storage';

export { FilterableFieldDescriptor } from './metadata-storage';

/** @internal */
let storage: GraphQLQueryMetadataStorage;
/** @internal */
export function getMetadataStorage(): GraphQLQueryMetadataStorage {
  if (!storage) {
    storage = new GraphQLQueryMetadataStorage();
  }
  return storage;
}
