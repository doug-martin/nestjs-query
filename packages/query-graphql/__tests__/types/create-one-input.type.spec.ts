import { plainToClass } from 'class-transformer';
import { MinLength, validateSync } from 'class-validator';
import { Resolver, Query, Int, Args, InputType, Field } from '@nestjs/graphql';
import { CreateOneInputType } from '../../src';
import { generateSchema } from '../__fixtures__';

describe('CreateOneInputType', (): void => {
  @InputType()
  class FakeType {
    @Field()
    @MinLength(5)
    field!: string;
  }
  @InputType()
  class CreateOne extends CreateOneInputType('fakeInput', FakeType) {}

  it('should create an args type with the field as the type', async () => {
    @Resolver()
    class CreateOneInputTypeSpec {
      @Query(() => Int)
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      test(@Args('input') input: CreateOne): number {
        return 1;
      }
    }
    const schema = await generateSchema([CreateOneInputTypeSpec]);
    expect(schema).toMatchSnapshot();
  });

  it('should properly assign the input field', () => {
    const input = { field: 'hello' };
    const it = plainToClass(CreateOne, { input });
    expect(it.input).toEqual(input);
    expect(it.input).toBeInstanceOf(FakeType);
  });

  it('should assign the typeName to the input field', () => {
    const input = { field: 'hello' };
    const it = plainToClass(CreateOne, { fakeInput: input });
    expect(it.input).toEqual(input);
    expect(it.input).toBeInstanceOf(FakeType);
  });

  describe('validation', () => {
    it('should validate the input property', () => {
      const input = { field: 'hola' };
      const it = plainToClass(CreateOne, { input });
      const errors = validateSync(it);
      expect(errors).toEqual([
        {
          children: [
            {
              children: [],
              constraints: {
                minLength: 'field must be longer than or equal to 5 characters',
              },
              property: 'field',
              target: input,
              value: input.field,
            },
          ],
          property: 'input',
          target: { input },
          value: input,
        },
      ]);
    });

    it('should assign the typeName to the input field', () => {
      const input = { field: 'hola' };
      const it = plainToClass(CreateOne, { fakeInput: input });
      const errors = validateSync(it);
      expect(errors).toEqual([
        {
          children: [
            {
              children: [],
              constraints: {
                minLength: 'field must be longer than or equal to 5 characters',
              },
              property: 'field',
              target: input,
              value: input.field,
            },
          ],
          property: 'input',
          target: { input },
          value: input,
        },
      ]);
    });
  });
});
