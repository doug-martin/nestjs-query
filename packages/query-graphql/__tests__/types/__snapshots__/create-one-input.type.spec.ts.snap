// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateOneInputType should create an args type with the field as the type 1`] = `
type Query {
  test(input: CreateOne!): Int!
}

input CreateOne {
  """The record to create"""
  fakeInput: FakeType!
}

input FakeType {
  field: String!
}

`;
